configuration/scripts/ratos-logging.sh

```
@@ -0,0 +1,238 @@
+    
+    # Check if we should log this level
+    local level_value=${LOG_LEVELS[$level]}
+    if [[ $level_value -lt $CURRENT_LOG_LEVEL ]]; then
+        return 0
+    fi
+    
+    # Rotate log if needed
+    rotate_log_if_needed
+    
+    # Build JSON log entry
+    local timestamp=$(get_timestamp)
```

Shellcheck (0.10.0)

[warning] Declare and assign separately to avoid masking return values.

(SC2155)